service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java

```diff
@@ -939,4 +944,41 @@ public static void checkException(SdkException ex, String altMessage, boolean ig
 
     throw new ApiException(altMessage, ex);
   }
+
+  /**
+   * Delete the Security Group associated with a Workspace.
+   *
+   * @param crlClientConfig CRL client configuration
+   * @param credentialsProvider Credentials provider for the WSM IAM role
+   * @param workspaceUuid UUID of the Workspace associated with the Security Group
+   * @param awsRegion AWS Region the Security Group exists in
+   * @param securityGroupId ID of Security Group to delete
+   * @throws {@link NoSuchElementException} if a Security Group with the passed ID does not exist in
+   *     the passed region
+   */
+  public static void deleteWorkspaceSecurityGroup(
+      ClientConfig crlClientConfig,
+      AwsCredentialsProvider credentialsProvider,
+      UUID workspaceUuid,
+      Region awsRegion,
+      String securityGroupId) {
+    try (EC2SecurityGroupCow regionCow =
+        EC2SecurityGroupCow.instanceOf(crlClientConfig, credentialsProvider, awsRegion)) {
+      regionCow.delete(securityGroupId);
+
+      logger.info(
+          "Deleted Security Group ID '{}' for Workspace {} (Landing Zone {})",
+          securityGroupId,
+          workspaceUuid.toString(),
+          awsRegion.toString());
+
+    } catch (Ec2Exception e) {
+      if (e.awsErrorDetails().errorCode().equals("InvalidGroup.NotFound")) {
+        throw new NoSuchElementException(
```

Comment:
This is specifically here to hide the AWS internals of parsing the AWS error code so that callers can be explicitly catch/log/ignore `NoSuchElementException` for idempotent deletes, and any other AWS exceptions will propagate.  Non-idempotent deletes (rollback case) will let the NSE propagate.